fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBStoreBase.java

```diff
@@ -80,9 +89,22 @@ public SubspaceProvider getSubspaceProvider() {
         return subspaceProvider;
     }
 
+    @Nonnull
+    public CompletableFuture<Subspace> getSubspaceAsync() {
+        if (subspaceFuture == null) {
+            subspaceFuture = subspaceProvider.getSubspaceAsync(context).whenComplete((s, e) -> {
```

Comment:
Hm, so if `getSubspaceAsync` returns an exceptional future, I think this will cache the exceptional future forever. I think the `whenComplete` could set `subspaceFuture` to `null`, except then there is a race condition that might result in this function returning `null`. Maybe returning an exceptional future in that case is fine? The value is only cached for the lifetime of an `FDBStoreBase`, which can't exceed one transaction's time.